src/blocks/plugins/feedback/feedback-form.tsx

```diff
@@ -0,0 +1,161 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	useEffect,
+	useState
+} from '@wordpress/element';
+import {
+	Button,
+	Spinner,
+	TextareaControl
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal Dependencies
+ */
+import './editor.scss';
+
+const { version } = window.themeisleGutenberg ? window.themeisleGutenberg : window.otterObj;
+
+const collectedInfo = [
+	{
+		name: __( 'Plugin version',  'otter-blocks' ),
+		value: version
+	},
+	{
+		name: __( 'Feedback', 'otter-blocks' ),
+		value: __( 'Text from the above text area', 'otter-blocks' )
+	}
+];
+
+/**
+ * Displays a button that opens a modal for sending feedback
+ *
+ * @param source The area from where the feedback is given
+ * @param status
+ * @param setStatus
+ * @returns
+ */
+const FeedbackForm = ({
+	source,
+	status,
+	setStatus
+}): JSX.Element => {
+	const [ feedback, setFeedback ] = useState( '' );
+	const [ showInfo, setShowInfo ] = useState( false );
+
+	useEffect( () => {
+		const info = document.querySelector( '.o-feedback-form .info' ) as HTMLElement;
+		if ( info ) {
+			info.style.height = showInfo ? `${ info.querySelector( '.wrapper' ).clientHeight }px` : '0';
+		}
+
+	}, [ showInfo ]);
+
+	const sendFeedback = () => {
+		const trimmedFeedback = feedback.trim();
+		if ( 5 >= trimmedFeedback.length ) {
+			setStatus( 'emptyFeedback' );
+			return;
+		}
+
+		setStatus( 'loading' );
+		fetch( 'https://api.themeisle.com/tracking/feedback', {
+			method: 'POST',
+			headers: {
+				'Content-Type': 'application/json'
+			},
+			body: JSON.stringify({
+				slug: 'otter-blocks',
+				version: version,
+				feedback: trimmedFeedback,
+				data: {
+					'feedback-area': source
+				}
+			})
+		}).then( r => {
+			if ( ! r.ok ) {
+				setStatus( 'error' );
+				return;
+			}
+
+			setStatus( 'submitted' );
+		});
```

Comment:
A practical example is this. The submit button is still showing the loading state instead of an error.

![image](https://user-images.githubusercontent.com/17597852/191479171-ce8698e7-0a30-495f-bc1f-98318a40e46c.png)